### ISSUE TITLE
Snakemake deletes input directory in dry-run

### ISSUE THREAD

**Issue by baraaorabi** [snakemake/snakemake#480]
**Snakemake version**
v5.19.3 (installed via conda)

**Describe the bug**
One of the rules has a directory as input.
When running Snakemake in dry mode (`-n` flag), Snakemake deletes the input directory.

**Logs**
```bash
(conda) baraa@machine:~$ mkdir test
(conda) baraa@machine:~$ ll test
total 0
drwxrwxrwx 1 baraa baraa 4096 Jul  7 12:08 ./
drwxr-xr-x 1 baraa baraa 4096 Jul  7 12:08 ../
(conda) baraa@machine:~$ snakemake -np
Building DAG of jobs...
Job counts:
        count   jobs
        1       all
        1       mkdir
        2

[Tue Jul  7 12:09:05 2020]
rule mkdir:
    input: test
    output: test/testing
    jobid: 1

mkdir test/testing

[Tue Jul  7 12:09:05 2020]
localrule all:
    input: test/testing
    jobid: 0

Job counts:
        count   jobs
        1       all
        1       mkdir
        2
This was a dry-run (flag -n). The order of jobs does not reflect the order of execution.
(conda) baraa@machine:~$ snakemake -np
Building DAG of jobs...
MissingInputException in line 5 of /home/baraa/Snakefile:
Missing input files for rule mkdir:
test
(conda) baraa@machine:~$ ll test
ls: cannot access 'test': No such file or directory
```

**Minimal example**

```python
rule all:
    input:
        'test/testing'

rule mkdir:
    input:
        'test'
    output:
        directory('test/testing')
    shell:
        'mkdir {output}'
```
**Additional context**

**Comment by baraaorabi** [comment]
Note: if the input folder is not empty, it seems that Snakemake does not delete it

**Comment by jmeppley** [comment]
This bug may have been added with commit 84c545856613f358ed2e8cca8c7cb11bfd755897 by @johanneskoester. 

It looks like job.empty_remote_dirs() became job.empty_dirs() and it carried over the behavior of finding empty input dirs but now it's being applied to all inputs, not just remotes. I could see how it would make sense to clean up empty remote input dirs, but not other inputs? 

We may want to qualify the cleanup_workdkir method to only delete empty input dirs if they are for remote jobs?

**Comment by johanneskoester** [comment]
Fixed in commit becd36ca.

**Comment by johanneskoester** [comment]
Thanks for reporting!
